tangelo/tangelo/__init__.py

```diff
@@ -145,8 +148,13 @@ def good(path):
     # to Nones will have been logged).
     newpaths = filter(lambda x: x is not None, map(good, runtimepaths))
 
+    # Use the import lock to have some thread safety
+    imp.acquire_lock()
+    # Exclude paths we've already added to the system
+    newpaths = [path for path in newpaths if path not in sys.path]
```

Comment:
Historically (at least up through Python 2.5), comprehensions were vastly faster than lambdas (one or two orders of magnitude in many instances).  I don't know if this is still true in Python 2.7/3.4, but since I came up through older versions, I'm now lambda shy.